aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1128,5 +1130,136 @@ Tensor _sparse_csr_prod_cpu(const Tensor& input, IntArrayRef dims_to_reduce, boo
   return result;
 }
 
+Tensor& _compressed_row_strided_mm_out(const Tensor& compressed, const Tensor& strided, Tensor& result) {
```

Comment:
This function also accept the scalars alpha/beta and use them in the same way as  `at::native::sparse::impl::cuda::spmm`, 
`at::native::sparse::impl::cuda::block_sparse_mm`, and `at::native::sparse::impl::mkl::addmm_sparse_input_dense_result` do. These functions are specific to the pattern `(sparse x dense -> dense)` and call into the backend libraries for supported datatypes and they should use the function as a fallback when the data type is not supported.  

All of those options perform `C = \alpha( A @ B) + \beta C`, which this interface should as well. 

NIT: I think this should function should live in `SparseBlasImpl.cpp` and be a member of the namespace `at::native::sparse::impl`. The backend specific functions for this and similar operations live in `mkl/SparseBlasImpl.cpp` and its counterpart `cuda/SparseBlasImpl.cpp` and  the namespaces `at::native::sparse::impl::<backend>`. So it would make sense for this impl of mm like ops to follow suit, discarding the backend specific namespace layer and folder level as it is backend independent.

Reply:
I was really waiting for this comment :)

Reply:
I think it makes sense to enable it only for the dtypes which are not supported by the backend libraries, at that point it can be tested by adding the previously un-supported dtype(s) to the test decorators. 

At that point we can profile, identify any optimizations and discuss other situations where it might be a better choice than what is already in use.